packages/adblocker/src/engine/bucket/cosmetic.ts

```diff
@@ -345,16 +346,20 @@ export default class CosmeticFilterBucket {
   public getHtmlRules({
     domain,
     hostname,
+
+    isFilterExcluded,
   }: {
     domain: string;
     hostname: string;
+
+    isFilterExcluded: (filter: IFilter) => boolean;
   }): CosmeticFilter[] {
     // Tokens from `hostname` and `domain` which will be used to lookup filters
     // from the reverse index. The same tokens are re-used for multiple indices.
     const hostnameTokens = createLookupTokens(hostname, domain);
     const rules: CosmeticFilter[] = [];
     this.htmlIndex.iterMatchingFilters(hostnameTokens, (rule: CosmeticFilter) => {
-      if (rule.match(hostname, domain)) {
+      if (rule.match(hostname, domain) && (!isFilterExcluded || !isFilterExcluded(rule))) {
```

Comment:
having a default value returning false would simplify those calls

Reply:
The change was made from using default value to qualify the review from @remusao . Do you have any changes in mind?

Reply:
Refs https://github.com/ghostery/adblocker/pull/3782#discussion_r1488535561